app/models/record.server.ts

```diff
@@ -12,42 +13,71 @@ export async function getRecordById(id: Record['id']) {
   return prisma.record.findUnique({ where: { id } });
 }
 
-export async function createRecord(
-  data: Pick<Record, 'username' | 'name' | 'type' | 'value' | 'status'>
-) {
+export async function createRecord(data: Pick<Record, 'username' | 'type' | 'name' | 'value'>) {
   // Set expiration date 6 months from now
   const expiresAt = new Date();
   expiresAt.setMonth(expiresAt.getMonth() + 6);
+  const status = RecordStatus.pending;
 
-  return prisma.record.create({ data: { ...data, expiresAt } });
+  return prisma.record.create({ data: { ...data, expiresAt, status } });
 }
 
 export async function updateRecordById(
   id: Record['id'],
-  username?: Record['username'],
-  name?: Record['name'],
   type?: Record['type'],
+  name?: Record['name'],
   value?: Record['value'],
+  status?: Record['status'],
+  username?: Record['username'],
   description?: Record['description'],
   course?: Record['course'],
   ports?: Record['ports'],
-  expiresAt?: Record['expiresAt'],
-  status?: Record['status']
+  expiresAt?: Record['expiresAt']
 ) {
   return prisma.record.update({
     where: { id },
     data: {
-      username,
-      name,
       type,
+      name,
       value,
+      status,
+      username,
       description,
       course,
       ports,
       expiresAt,
+    },
+  });
+}
+
+export async function updateRecordStatusById(id: Record['id'], status: Record['status']) {
+  const now = new Date();
+  const expireToSet = new Date();
+  expireToSet.setMonth(now.getMonth() + 6);
```

Comment:
I would use `dayjs` to manipulate dates. See issue #311

Reply:
Great suggestion! Thanks